### PR TITLE
Add clock qos to node options

### PR DESCRIPTION
**Pull request by ivanpauno**
Fixes https://github.com/ros2/rclcpp/issues/1318.

I will open a similar PR in `rclpy`.

(edit) Edited title, as this is allowing to configure the clock qos and not the rosout qos.

### CODE REVIEW

File: rclcpp/src/rclcpp/node.cpp [hunk 1]
```diff
@@ -138,7 +138,8 @@ Node::Node(
       node_services_,
       node_logging_,
       node_clock_,
-      node_parameters_
+      node_parameters_,
+      options.clock_qos(),
```
Comment (wjwwood) [inline rclcpp/src/rclcpp/node.cpp:142]: Is the `,` going to work?

File: rclcpp/src/rclcpp/node_interfaces/node_time_source.cpp [hunk 2]
```diff
@@ -26,14 +26,16 @@ NodeTimeSource::NodeTimeSource(
...
+  const rclcpp::QoS & qos)
 : node_base_(node_base),
   node_topics_(node_topics),
   node_graph_(node_graph),
   node_services_(node_services),
   node_logging_(node_logging),
   node_clock_(node_clock),
-  node_parameters_(node_parameters)
+  node_parameters_(node_parameters),
+  time_source_(qos),
```
Comment (wjwwood) [inline rclcpp/src/rclcpp/node_interfaces/node_time_source.cpp:38]: Same here, I think C++ won't allow the trailing `,`.

**Review by wjwwood — CHANGES_REQUESTED** [review]

File: rclcpp/src/rclcpp/node.cpp (hunk 1 above)
Comment (ivanpauno) [reply rclcpp/src/rclcpp/node.cpp:142]: Sorry, I compiled and tested the wrong workspace :man_facepalming:

Comment (ivanpauno) [reply rclcpp/src/rclcpp/node.cpp]: fixed

**Review by wjwwood — APPROVED** [review]
lgtm

**Review by fujitatomoya — APPROVED** [review]
lgtm

**Comment by ivanpauno** [comment]
resolved conflicts

**Comment by ivanpauno** [comment]
CI (rclcpp, test_rclcpp):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12633/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7594/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10338/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12562/)

**Comment by nnmm** [comment]
@ivanpauno I thought I asked before, but I can't see my earlier comment now, apologies if this is a duplicate. I hope you can help me understand the following things about this PR:
* Is the QoS profile for the clock subscription ever used? It seems to me like it should replace the hardcoded QoS in `rclcpp/src/rclcpp/time_source.cpp:239`, shouldn't it?
* Is it intentional that `rclcpp/include/rclcpp/node_interfaces/node_time_source.hpp:51` mentions `RosoutQos` and not `ClockQos`?

**Comment by wjwwood** [comment]
@nnmm, in the future, can you provide github links rather than file/line number, it makes it easier to know we're speaking about the same thing.

For example, on the master branch the first thing you mentioned doesn't seem to point to what you're talking about:

https://github.com/ros2/rclcpp/blob/f3a51877758f3bd5a8300b1cfb6d3ad85cf7257e/rclcpp/src/rclcpp/time_source.cpp#L239

So I'm not sure which branch/version to reference. You can also link/reference lines in this pull request's changes, if that's what you're talking about.

For the second, that reference looks ok still:

https://github.com/ros2/rclcpp/blob/f3a51877758f3bd5a8300b1cfb6d3ad85cf7257e/rclcpp/include/rclcpp/node_interfaces/node_time_source.hpp#L51

@ivanpauno I'm not sure about this either, can you comment?

**Comment by nnmm** [comment]
@wjwwood Ah, yes, the line number refer to the file after these changes.

**Comment by nnmm** [comment]
@ivanpauno Could you comment on this? For clarity, the lines I am referring to above are https://github.com/ros2/rclcpp/blob/3b96b1cd3a4b1b0f700b28bca40814a9eae45ea9/rclcpp/src/rclcpp/time_source.cpp#L239 and https://github.com/ros2/rclcpp/blob/3b96b1cd3a4b1b0f700b28bca40814a9eae45ea9/rclcpp/include/rclcpp/node_interfaces/node_time_source.hpp#L51.

**Pull request merged**